pkg/bgpv1/types/bgp.go

```diff
@@ -57,6 +57,102 @@ type PathResponse struct {
+}
+
+// PolicyConditions represent conditions of a policy statement.
+//
+// +deepequal-gen=true
+type PolicyConditions struct {
+	// MatchNeighbors matches ANY of the provided BGP neighbor IP addresses. If empty matches all neighbors.
+	MatchNeighbors []string
+	// MatchPrefixes matches ANY of the provided prefixes. If empty matches all prefixes.
+	MatchPrefixes []*PolicyPrefixMatch
+}
+
+// PolicyRouteAction defines the action taken on a route matched by a routing policy.
+type PolicyRouteAction int
+
+const (
+	// RouteActionNone does not affect processing of a route.
+	// The policy evaluation continues with the next policy statements / other policies.
+	RouteActionNone PolicyRouteAction = iota
+	// RouteActionAccept accepts a route into the RIB / adjacency RIB.
+	// No further policy statements / policies are evaluated for the route.
+	RouteActionAccept
+	// RouteActionReject rejects a route from the RIB / adjacency RIB.
+	// No further policy statements / policies are evaluated for the route.
+	RouteActionReject
+)
+
+// PolicyActions define policy actions taken on route matched by a routing policy.
+//
+// +deepequal-gen=true
+type PolicyActions struct {
+	// RouteAction defines an action taken on the matched route.
+	RouteAction PolicyRouteAction
+	// AddCommunities defines a list of BGP community strings to be added to the matched route.
+	// If empty, no communities will be added.
+	AddCommunities []string
+	// SetLocalPreference define a BGP local preference value to be set on the matched route.
+	// If nil, no local preference is set.
+	SetLocalPreference *int64
+}
+
+// PolicyStatement represents a single statement of a routing Policy. It contains conditions for
+// matching a route and actions taken if a route matches the conditions.
+//
+// +deepequal-gen=true
+type PolicyStatement struct {
+	// Conditions of the statement. If ALL of them match a route, the Actions are taken on the route.
+	Conditions PolicyConditions
+	// Actions define actions taken on a matched route.
+	Actions PolicyActions
+}
+
+// PolicyType defines the type of routing policy.
+type PolicyType int
+
+const (
+	// PolicyTypeExport represents export routing policy type (affecting how the routes from RIB are advertised to peers).
+	PolicyTypeExport PolicyType = iota
+	// PolicyTypeImport represents import routing policy type (affecting how the routes are imported into RIB).
+	PolicyTypeImport
+)
+
+// Policy represents a BGP routing policy, also called "route map" in some BGP implementations.
+// It can contain multiple Statements that are evaluated in the given order. Each Statement
+// contains conditions for matching a route and actions taken if a route matches the conditions.
+// Whenever a Statement matches a route and the action taken on it is to either accept or reject the route,
+// the policy evaluation for the given route stops, and no further Statements nor other Policies are evaluated.
+//
+// +deepequal-gen=true
+type Policy struct {
```

Comment:
Consider naming this 'RoutePolicy'? Since you seem to refer to it by this name in most comment strings.

Reply:
Renamed to `RoutePolicy`